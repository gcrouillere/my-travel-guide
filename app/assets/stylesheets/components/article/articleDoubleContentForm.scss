.article-container-form {
  .double-content-elements {
    display: flex;
    justify-content: center;
    border: 1px solid #ccc;
    border-radius: 2px;
    .photoInput, .mapInput, .textContentInput {
      margin: 0;
      width: 50%;
    }
    .photoInput {
      .mixedPhotoMobile {
        display: none;
      }
    }
    .mapInput {
      .mapBloc {
        height: 100%;
        display: flex;
        flex-direction: column;
        .googleMap {
          height: auto;
          flex: 1 0 auto;
        }
      }
    }
    .textContentInput {
      display: flex;
      flex-direction: column;
      .quill {
        height: 100%;
        display: flex;
        flex-direction: column;
        .ql-toolbar.ql-snow {
          height: auto;
        }
        .ql-container.ql-snow {
          height: auto;
          flex: 1 0 auto;
        }
      }
    }
  }
  .double-content-elements > div:nth-of-type(2) {
    border-right: 1px solid #ccc;
  }

  @media all and (max-width:500px) {
    .elementResize {
      display: none;
    }
    .double-content-elements {
      display: block;
      height: auto !important;
      .photoInput, .mapInput, .textContentInput {
          width: 100%;
      }
      .photoInput {
        background-image: none !important;
        .mixedPhotoMobile {
         display: block;
        }
      }
    }
  }
}
