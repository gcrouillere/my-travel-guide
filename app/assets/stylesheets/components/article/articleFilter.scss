.articlesFilter .audienceFilter {
  .form-check {
    label {
      padding: 5px;
    }
  }
}

.audienceFilter.audienceSelection {
  margin: 10px auto;
}

.stringFilter {
  position: relative;
  width: 52%;
  @media all and (max-width:768px) {
    width: 82%;
  }
  margin: 10px auto 30px auto !important;
  .stringFilterInput {
    padding: 5px;
    outline: none;
    border-radius: 30px;
    width: 100%;
    text-align: center;
    border: 1px solid #ccc;
    text-shadow: 0px 1px 0px rgba(0,0,0,0.2);
    background: white;
  }
}

.stringFilter.ticked span:after {
  position: absolute;
  top: -4px; bottom: -4px;
  left: -4px; right: -4px;
  background: linear-gradient(to right, #343a40, #00bf8f);
  opacity: 1;
  content: '';
  z-index: -1;
  border-radius: 30px;
  transition: all 0.3s ease-in-out;
}
.stringFilter.blank span:after {
  position: absolute;
  top: -4px; bottom: -4px;
  left: -4px; right: -4px;
  background: linear-gradient(to right, #343a40, #00bf8f);
  opacity: 0;
  content: '';
  z-index: -1;
  border-radius: 30px;
  transition: all 0.3s ease-in-out;
}

.filters {
  height: auto;
  display: flex;
  flex-direction: column;
  position: relative;
  .mapRow {
    height: 100%;
  }
  .mapRow.hide {
    display: none;
  }
  .googleMap {
    height: 100%;
    width: 100%;
  }
  .listSwitch {
    position: fixed;
    display: flex;
    flex-direction: column;
    z-index: 1;
    .switchView {
      margin: 8px;
      cursor: pointer;
      padding: 1px;
      width: 33px;
      height: 33px;
      border-radius: 50%;
      background: white;
      color: white;
      box-shadow: 0 0 20px rgba(0,0,0,.2);
      text-shadow: 1px 1px 1px rgba(0,0,0,0.6);
      border: 2px solid white;
    }
  }
}
.filters.full {
  height: calc(100vh - 76px);
  padding-bottom: 20px;
}
.listOnMap {
  width: 100%
}
