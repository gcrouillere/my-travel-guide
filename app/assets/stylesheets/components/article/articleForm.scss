.article-container {
  padding-right: 30px !important;
  padding-left: 30px !important;

  .audienceSelection {
    margin: 30px 0;
  }

  .blocAddition {
    margin: 30px 0;
  }

  header {
    margin: 30px 0;
    input {
      width: 100%;
      height: 28px;
      border-radius: 2px;
      padding: 25px;
      border: 1px solid #ccc;
      transition: .3s;
      box-shadow: 1px 1px 0 rgba(241,241,241,0.35);
      font-size: 25px;
    }
  }

  .textContentInput {
    margin: 30px 0;
    .ql-container {
      border-radius: 0px 0px 2px 2px;

      .ql-toolbar {
        border-radius: 2px 2px 0px 0px;
      }

     .ql-editor {
        font-family: inherit;
        line-height: 22px;
      }
    }
  }

}
