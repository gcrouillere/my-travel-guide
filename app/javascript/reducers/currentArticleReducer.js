import { FETCH_ARTICLE } from '../actions'

export default function(state = null, action) {

  switch(action.type) {
    case FETCH_ARTICLE:
      return action.payload
    default:
      return state
  }
}
