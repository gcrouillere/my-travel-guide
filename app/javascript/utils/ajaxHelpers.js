import $ from 'jquery'
import { GOOGLEKEY } from './../config/config'

export default {


  async getDataFromURL(url) {
    let returnData = null

    await $.ajax({
      method: 'GET',
      url: url,
      dataType: "JSON"
    })
    .done(data => { returnData = data })
    .fail(data => console.log(data))

    return returnData
  },

  async updateDataInURL(url, data, token) {
    let returnData = null

     await $.ajax({
      method: "PUT",
      beforeSend: function(xhr) { xhr.setRequestHeader('X-CSRF-Token', token) },
      url: url,
      data: data
    })
    .done(data => { returnData = data })
    .fail(data => console.log(data))

    return returnData
  },

  async geocodePlace(adress) {
    const formattedAdress = adress.replace(/,/g, '').split(' ').join('+')
    let map = null

    await $.ajax({
      method: "GET",
      url: `https://maps.googleapis.com/maps/api/geocode/json?address=${formattedAdress}&key=${GOOGLEKEY.googleAPIBrowserKey}`
    }).done(data => {
      map = {
        lat: data.results[0].geometry.location.lat,
        lng: data.results[0].geometry.location.lng,
        name: data.results[0].formatted_address
    }})

    return map
  }

}
