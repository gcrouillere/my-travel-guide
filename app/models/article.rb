class Article < ApplicationRecord
  has_many :text_contents, -> { order('position ASC') }, dependent: :destroy
  has_many :maps, -> { order('position ASC') }, dependent: :destroy
  has_many :photos, -> { order('position ASC') }, dependent: :destroy
  has_many :double_contents, -> { order('position ASC') }, dependent: :destroy
  has_many :article_audience_selections
  has_many :audience_selections, through: :article_audience_selections
  belongs_to :user

  include AlgoliaSearch

  algoliasearch auto_index: false, auto_remove: false do
    attribute :title
    add_attribute :article_id
  end

  def elements_position_mapping
    self.as_json(include: {
        text_contents: { methods: :class_name },
        maps: { methods: :class_name },
        photos: { methods: :class_name },
        double_contents: { methods: :class_name },
      }).select {|k, v| k.match(/^maps$|^text_contents$|^photos$|^double_contents$/)}
      .select {|k, v| v != []}
      .map {|k, v| v.map {|e| [e["class_name"], e["id"], e["position"]] }}
      .flatten(1)
  end

  def article_id
    self.id
  end
end
