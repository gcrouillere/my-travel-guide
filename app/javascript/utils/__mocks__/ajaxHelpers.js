import update from 'immutability-helper'
import article from './../../../../test/javascript/__mocks__/fakeArticle.json'
import audienceSelection from './../../../../test/javascript/__mocks__/audienceSelection.json'
import updatedArticle from './../../../../test/javascript/__mocks__/fakeArticleFedWithText.json'

const map = article.maps[0]

const marker = map.markers[0]

const polyline = map.polylines[0]

const photo = article.photos[0]

const ajaxCall = jest.fn((method, url, data, token) => {
  if ( /^\/articles\/[0-9]+/.test(url) ) {
    if (method == 'GET') { return Promise.resolve(article) }
    if  (method == 'PUT') {
      if (data.article.audience_selection_ids && data.article.audience_selection_ids.length ==  2) {
        return Promise.resolve(updatedArticle)
      } else {
        return Promise.resolve(article)
      }
    }
  }

  if (/^\/audience_selections/.test(url) ) {
    if (method == 'GET') { return Promise.resolve(audienceSelection) }
  }

  if ( /^\/maps/.test(url) ) {
    if  (method == 'PUT') {
      if (data.map.zoom) {
        let newMap = update(map, {zoom: {$set: 2}})
        return Promise.resolve(newMap)
      } else if (data.map.height) {
        let newMap = update(map, {height: {$set: 339}})
        return Promise.resolve(newMap)
      } else if (data.map.name) {
        let newMap = update(map, {name: {$set: data.map.name}})
        return Promise.resolve(newMap)
      } else if (data.map.lat) {
        let newMap = update(map, {lat: {$set: 1}, lng: {$set: 1}})
        return Promise.resolve(newMap)
      } else if (data.map.show_map_center_as_marker != null) {
        let newMap = update(map, { show_map_center_as_marker: { $set: data.map.show_map_center_as_marker }})
        return Promise.resolve(newMap)
      }
    }
  }

  if ( /^\/markers/.test(url) ) {
    if  (method == 'POST') {
      if (data.marker.position) {
        let newMarker = update(marker, {id: {$set: 1}, position: {$set: 3}})
        return Promise.resolve(newMarker)
      } else {
        let newMarker = update(marker, {id: {$set: 1}, lat: {$set: 6}, lng: {$set: 6}})
        return Promise.resolve(newMarker)
      }
    }
    if  (method == 'PUT') {
      if (data.marker.description) {
        let newMarker = update(marker, {description: {$set: "my description"}})
        return Promise.resolve(newMarker)
      }
      if (data.marker.logo) {
        let newMarker = update(marker, {logo: {$set: "restaurantLogo"}})
        return Promise.resolve(newMarker)
      }
      if (data.marker.lat) {
        let newMarker = update(marker, {lat: {$set: 69}, lng: {$set: 69}})
        return Promise.resolve(newMarker)
      }
    }
    if  (method == 'DELETE') { return Promise.resolve(marker) }
  }

  if ( /^\/polylines/.test(url) ) {
    if  (method == 'GET') {
      let newPolyline = update(polyline, {id: {$set: 1}})
      return Promise.resolve(newPolyline)
    }
    if  (method == 'PUT') {
      if (data.polyline.distance_displayed) {
        let newPolyline = update(polyline, {distance_displayed: {$set: data.polyline.distance_displayed}})
        return Promise.resolve(newPolyline)
      }
    }
    if  (method == 'DELETE') {
      return Promise.resolve(polyline)
    }
  }

  if(/^\/photos/.test(url)) {
    if (method == 'PUT') {
      let newPhoto = update(photo, {css_width: {$set: 30}})
      return Promise.resolve(newPhoto)
    }
  }
})

const geocodePlace = jest.fn((adress) => { return Promise.resolve({name: adress}) })

module.exports = {
  ajaxCall,
  geocodePlace
}
