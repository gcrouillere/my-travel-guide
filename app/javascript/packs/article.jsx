import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import $ from 'jquery'
import update from 'immutability-helper'

import ajaxHelpers from './../utils/ajaxHelpers'
import mapHelper from './../utils/mapHelper'
import photoHelpers from './../utils/photoHelpers'
import orderHelper from './../utils/articleContentHelper'
import markerLogos from './articleForm/mapForm/markerLogos'
import ShowSecondContentButton from './articleForm/formElementManagement/showSecondContentButton'

export class Article extends Component {
  constructor(props) {
    super(props)
    this.state = {
      articleElements: [],
      maps: [],
      texts: [],
      title: "",
      clientWidth: document.body.clientWidth,
      doubleContentActives: {}
    }
    this.textContentDivs = {}
    this.mapDivs = {}
    this.articleContent = React.createRef()
  }

  componentDidUpdate(prevProps) {
    if (this.checkUpdate(prevProps, this.props)) {
      this.setState({
        articleElements: orderHelper.orderArticleElements(this.props.currentArticle),
        maps: this.concatContentList("maps"),
        texts: this.concatContentList("text_contents"),
        title: this.props.currentArticle.title,
        doubleContentActives: this.doubleContentInitialStates()
      }, () => {
        this.createMaps()
        this.createTextContentsHTML()
      })
    }
  }

  doubleContentInitialStates = () => {
    let doubleContentActives = this.state.doubleContentActives
    this.props.currentArticle.double_contents.forEach( x => {
      const key = `doubleContent-${x.id}`
      doubleContentActives = update(doubleContentActives, {[key]: { $set: false }})
    })
    return doubleContentActives
  }

  checkUpdate(prevProps, newProps) {
    if (!prevProps.currentArticle && newProps) return true
    if (prevProps.currentArticle) {
      if (prevProps.currentArticle.id !== newProps.currentArticle.id) return true
    }
    if (newProps.currentArticle.title &&
        newProps.currentArticle.title != "" &&
        this.articleContent.current.innerText === "") return true
    return false
  }

  concatContentList = (content) => {
    let articleContent = this.props.currentArticle[content]
    let doubleContentsContent = this.props.currentArticle.double_contents.reduce((acc, x) => acc.concat(x[content]), [])
    return articleContent.concat(doubleContentsContent)
  }

  createMaps() {
    Object.keys(this.mapDivs).map(mapID => {
      const map = this.state.maps.find( x => x.id === parseInt(mapID))
      const googleMap = this.createMap(map, mapID)

      let markers = map.markers ||Â []
      this.addCenter(map, markers)
      markers.forEach( marker => {

        const icon = markerLogos[marker.logo].url
        const googleMarker = mapHelper.createMarker(marker, googleMap, icon, null, false)
        if (marker.description) {
          const infowindow = new google.maps.InfoWindow({content: marker.description, disableAutoPan: true});
          infowindow.open(googleMap, googleMarker);
        }
      })

      const polylines = map.polylines || []
      polylines.forEach( polyline => {

       const googlePolyline = this.createPolyline(polyline, googleMap)
       polyline.markers.forEach((marker, index) => {

        let icon = mapHelper.getPolylinePointLogo(index, polyline.markers)
        let tempMarker = mapHelper.createMarker(marker, googleMap, icon, index)
        mapHelper.manageDistanceInfoWindow(index, polyline.distance_displayed, polyline.markers,
        googlePolyline, tempMarker, googleMap)
       })
      })
    })
  }

  createMap(map, mapID) {
    return new google.maps.Map( this.mapDivs[mapID], {
      center: { lat: map.lat, lng: map.lng },
      zoom: map.zoom,
      mapTypeControl: true,
      zoomControl: true,
      zoomControlOptions: { position: google.maps.ControlPosition.LEFT_CENTER },
      scaleControl: false,
      streetViewControl: false,
      fullscreenControl: false,
      styles: [ { "featureType": "poi.business", "stylers": [{ "visibility": "off" }] } ]
    });
  }

  createPolyline(polyline, googleMap) {
    return new google.maps.Polyline({
      map: googleMap,
      path: polyline.markers,
      strokeColor: '#495057',
      strokeOpacity: 1.0,
      strokeWeight: 2,
      geodesic: true
    })
  }

  addCenter(map, markers) {
    if (map.show_map_center_as_marker) markers.unshift({
      lat: map.lat,
      lng: map.lng,
      description: map.name,
      logo: "markerLogo",
      mapCenter: true
    })
  }

  createTextContentsHTML() {
    Object.keys(this.textContentDivs).map(textContentID => {
      const text = this.state.texts.find( x => x.id === parseInt(textContentID) ).text
      this.textContentDivs[textContentID].insertAdjacentHTML('beforeend', text)
    })
  }

  defineHeight = (map) => {
    return document.body.clientWidth < 768 ? `${map.height}px` : "auto%"
  }

  definePhotoWidth = (photo) => { return this.state.clientWidth >= 768 ? photo.css_width : 100}

  activateSecondContent = (doubleContentID) => {
    let doubleContentActives = this.state.doubleContentActives
    let initialState = doubleContentActives[`doubleContent-${doubleContentID}`]
    doubleContentActives = update(doubleContentActives, { [`doubleContent-${doubleContentID}`]: { $set: !initialState }})
    this.setState({ doubleContentActives: doubleContentActives })
  }

  render() {
    console.log("render")
    console.log(this.state.doubleContentActives)
    return (
      <div className="container article-container">
        <div className="row">
           <div className="col">

            <header ref={this.articleContent}>
              <h1 className="article-header">{this.state.title}</h1>
            </header>

            <div className="article-content">
              {this.state.articleElements.map(element => {

                if (element.class_name == "TextContent") {
                  return <div key={`text-${element.id}`}
                  ref={div => this.textContentDivs[element.id] = div} className="textContent"></div>
                }

                if (element.class_name == "Map") {
                  return <div key={`map-${element.id}`} id={`map-${element.id}`}
                  ref={div => this.mapDivs[element.id] = div}
                  className="googleMap" style={{ width: '100%', height: `${element.height}px` }}></div>
                }

                if (element.class_name == "Photo") {
                  return <div className="photo" key={`photo-${element.id}`}>
                    <img src={ photoHelpers.retrievePhotoSRC(element) } alt={ element.original_filename }
                    style={{width: `${this.definePhotoWidth(element)}%`}} />
                    { element.display_title && <p className="photo-title">{ element.original_filename }</p> }
                  </div>
                }

                if (element.class_name == "DoubleContent") {
                  return (
                      <div className={`doubleContent
                        ${this.state.doubleContentActives ? (this.state.doubleContentActives[`doubleContent-${element.id}`] ? "active" : "") : null}`}
                       key={`doubleContent-${element.id}`} style={{minHeight: `${element.height}px`}}>
                        <ShowSecondContentButton activateSecondContent={this.activateSecondContent} text={"Show second content"}
                          id={element.id} />
                      { element.associated_instances_mapping.map( (type, index) => {
                        if (type[0] === "TextContent") {
                          const text_content = element.text_contents[index] || element.text_contents[0]
                          return <div key={`text-${text_content.id}`}
                            className={`textContent ${text_content.position === 0 ? "firstContent" : "secondContent"}`}
                            ref={div => this.textContentDivs[text_content.id] = div} >
                          </div>
                        }
                        if (type[0] === "Map") {
                          const map = element.maps[index] || element.maps[0]
                          return <div key={`map-${map.id}`} id={`map-${map.id}`} ref={div => this.mapDivs[map.id] = div}
                            className={`googleMap ${map.position === 0 ? "firstContent" : "secondContent"}`}
                            style={{ height: `${this.defineHeight(map)}` }}>
                          </div>
                        }
                        if (type[0] === "Photo") {
                          const photo = element.photos[index] || element.photos[0]
                          return <div key={`photo-${photo.id}`}
                            className={`photo ${photo.position === 0 ? "firstContent" : "secondContent"}`}
                            style={{ backgroundImage: `url(${photo.url})`}}>
                              <img className="photoMobile"src={photo.url} alt=""/>

                          </div>
                        }
                      })}
                    </div>
                  )
                }
              })}
            </div>

          </div>
        </div>
      </div>
    )
  }
}

Article.propTypes = {
  match: PropTypes.object.isRequired
}

function mapStateToProps(state) {
  return {
    currentArticle: state.currentArticle
  }
}

export default withRouter(connect(mapStateToProps, null)(Article))
