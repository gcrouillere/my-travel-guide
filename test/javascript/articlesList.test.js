import React, { Component } from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router, Route } from 'react-router-dom'
import update from 'immutability-helper'

import ArticlesList from '../../app/javascript/packs/articlesList'
import article from './__mocks__/fakeArticle'
import ajaxHelpers from '../../app/javascript/utils/ajaxHelpers'

import currentUserReducer from '../../app/javascript/reducers/currentUserReducer'
import currentArticleReducer from '../../app/javascript/reducers/currentArticleReducer'

const reducers = combineReducers({
  currentUser: currentUserReducer,
  currentArticle: currentArticleReducer
})

configure({ adapter: new Adapter() });

const articleslist = [article, update(article, {id: {$set: 1}})]

jest.mock('../../app/javascript/utils/ajaxHelpers', () => jest.fn())

ajaxHelpers.ajaxCall = jest.fn(() => { return Promise.resolve(articleslist) })

describe('ArticlesList test Suite', () => {

  it('renders component', async () => {
    const articlesListTree = await mount(
      <Provider store={createStore(reducers)}>
        <Router><ArticlesList/></Router>
      </Provider>
      )

    expect(toJson(articlesListTree)).toMatchSnapshot()
  })
})
