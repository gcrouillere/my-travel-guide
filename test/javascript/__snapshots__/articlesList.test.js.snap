// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList test Suite renders component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(ArticlesList))>
        <Route>
          <Connect(ArticlesList)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <ArticlesList
              currentUser={null}
              dispatch={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div
                className="container articles-container"
              >
                <div
                  className="row"
                />
              </div>
            </ArticlesList>
          </Connect(ArticlesList)>
        </Route>
      </withRouter(Connect(ArticlesList))>
    </Router>
  </BrowserRouter>
</Provider>
`;
